### PR TITLE
[flutter_webview] Fix `allowsInlineMediaPlayback` ignored on iOS

### PR DESCRIPTION
**Pull request by cyanglaz**
This PR fixes the `allowsInlineMediaPlayback` flag ignored on iOS.
This PR also enables some tests that were failing before, as well as the test for `allowsInlineMediaPlayback`

Fixes: https://github.com/flutter/flutter/issues/73768
Fixes: https://github.com/flutter/flutter/issues/72572

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/webview_flutter/example/integration_test/webview_flutter_test.dart [hunk 1]
```diff
@@ -479,24 +482,37 @@ void main() {
...
       expect(isPaused, _webviewBool(false));
-    }, skip: true /* https://github.com/flutter/flutter/issues/72572 */);
+    });
 
     testWidgets('Video plays inline when allowsInlineMediaPlayback is true',
         (WidgetTester tester) async {
       Completer<WebViewController> controllerCompleter =
           Completer<WebViewController>();
       Completer<void> pageLoaded = Completer<void>();
+      // String message1;
```
Comment (stuartmorgan) [inline packages/webview_flutter/example/integration_test/webview_flutter_test.dart:492]: What does this mean (here and below)?

File: packages/webview_flutter/ios/Classes/FlutterWebView.m [hunk 2]
```diff
@@ -361,6 +359,18 @@ - (NSString*)applySettings:(NSDictionary<NSString*, id>*)settings {
                                     [unknownKeys componentsJoinedByString:@", "]];
 }
 
+- (void)applyConfigurationSetting:(NSDictionary<NSString*, id>*)settings
+                  inConfiguration:(WKWebViewConfiguration*)configuration {
```
Comment (stuartmorgan) [inline packages/webview_flutter/ios/Classes/FlutterWebView.m:363]: Nit: s/in/to/

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits

File: packages/webview_flutter/example/integration_test/webview_flutter_test.dart (hunk 1 above)
Comment (cyanglaz) [reply packages/webview_flutter/example/integration_test/webview_flutter_test.dart:492]: oops, some testing code not cleaned up :) Good catch! Thanks

File: packages/webview_flutter/ios/Classes/FlutterWebView.m (hunk 2 above)
Comment (cyanglaz) [reply packages/webview_flutter/ios/Classes/FlutterWebView.m]: Also  did s/Setting/Settings

**Pull request merged**